inbox/util/threading.py

```diff
@@ -16,25 +19,24 @@ def fetch_corresponding_thread(db_session, namespace_id, message):
     # to a message always has a similar subject. This is only
     # right 95% of the time.
     clean_subject = cleanup_subject(message.subject)
+
+    # XXX: It is much faster to sort client-side by message ID, we therefore
+    # use `contains_eager` and `outerjoin` instead of `joinedload` to fetch
+    # the messages by thread in no particular order. We also use a limit to
```

Comment:
I think it's worth mentioning that with `joinedload`, the order would come from the `_thread` relation's backref's `order_by`.